docs/tutorials/craft-a-snap.rst

```diff
@@ -0,0 +1,549 @@
+
+    mkdir ukuzama-pyfiglet
+    cd ukuzama-pyfiglet
+
+A snap is defined in a declarative ``snapcraft.yaml`` file, called the *project file*.
+By constructing the file key by key, we'll be building the snap's particulars. Snapcraft
+has an ``init`` command that spawns a template project file. Let's start with that:
+
+.. code-block:: bash
+
+    snapcraft init
+
+From here onward, we'll be working primarily in the project file. Open
+``snap/snapcraft.yaml`` in a text editor.
+
+
+Define the package information
+------------------------------
+
+The first section inside a snap project file is typically its package information,
+sometimes informally referred to as its metadata. This information tells both humans and
+machines about the snap, such as its purpose, authors, license, and so on. The comments
+in the template describe how to use these keys.
+
+Replace the top section with:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+
+    name: ukuzama-pyfiglet
+    base: core24
+    version: '0.1'
+    summary: pyfiglet is a program for making large letters out of ordinary text
+    description: |
+      pyfiglet is a full port of FIGlet (http://www.figlet.org/) into pure python. It takes
+      ASCII text and renders it in ASCII art fonts (like the title above, which is the
+      'block' font).
+
+      This snap is not endorsed by the pyfiglet project.
+
+As this is a personal snap, we prepended the project name with a user name. Replace
+``ukuzama`` with your own user name. You might encounter other snaps in the Snap Store
+with naming that follows this pattern -- it's the recommended format, to keep personal
+copies of snaps distinct from their originals.
+
+We didn't alter the ``base`` key, since we want our project to be built on top of the
+latest Ubuntu LTS release.
+
+Since we're packaging a project authored by someone else, we ought to respect their
+intentions and thinking in the description keys. So, we reused them. The ``summary`` is
+a short description with a hard limit of 79 characters, but nothing like that is present
+in the pyfiglet README. Instead, we sourced one from the upstream project at `figlet.org
+<https://www.figlet.org>`_. The fuller ``description`` key, which has no length limit,
+is taken from pyfiglet. We added a disclaimer about endorsement at the end.
+
+
+Define the target platforms
+---------------------------
+
+As we're building on an x64 system, and we're only running basic tests for this
+tutorial, we should constrain our snap to only build on the current CPU architecture,
+AMD64. At a later point, when you're able to test on other platforms, you could widen
+the coverage in this configuration.
+
+Add the ``platform`` key after the project information:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+
+    platforms:
+      amd64:
+
+With this delcaration, Snapcraft will only build the snap on AMD64 machines, for AMD64
+machines. Take care to preserve the colon (:) in ``amd64:``.
```

Comment:
Might be worth cutting.

Reply:
If you're trying to shrink the tutorial, it could be dropped but I like it.  `platforms` is a confusing subject so a tutorial is a good place to start.